Entities/Common/FireScripts/FireAnim.as

```diff
@@ -30,17 +30,19 @@ void onInit(CSprite@ this)
 
 void onTick(CSprite@ this)
 {
-	this.getCurrentScript().tickFrequency = 24; // opt
+	this.getCurrentScript().tickFrequency = 24;
 	CBlob@ blob = this.getBlob();
 	CSpriteLayer@ fire = this.getSpriteLayer("fire_animation_large");
+	
 	if (fire !is null)
 	{
 		//if we're burning
 		if (blob.hasTag(burning_tag))
 		{
 			this.getCurrentScript().tickFrequency = 12;
 
-			fire.SetVisible(true);
+			if (!fire.isVisible())
+				fire.SetVisible(true);
```

Comment:
Why is this change required? The added `if` statement looks redundant.

Reply:
It is not required, I just figured it would be good practise to do it this way.
I tested with
```
	for (int z = 0; z < 500000; z++)
	{
		if (!fire.isVisible())
			fire.SetVisible(true);
	}
```
I barely could see any lag and I couldn't spot a difference in performance between having `if (!fire.isVisible())` vs. not having it. That also means there is no gain in having `if (!fire.isVisible())`. Should I revert it?

Reply:
It is indeed not useful because IIRC it is just a mere boolean flag in engine. Some other things, like SetLight, involve more involved logic when the state is _actually_ changed, but the check is usually redundant anyway.

Reply:
Generally speaking the bottleneck in this kind of call-heavy scripts is the FFI (i.e. calling engine functions from the script VM) in which case causing more calls could actually lower performance. Then again, that code is really not a hot loop, so it doesn't matter much. That is especially since it ticks at a throttled rate, which skips script execution completely, and while not free, it is probably as much of a performance save as you can get.